lib/praxis/multipart/parser.rb

```diff
@@ -87,9 +88,13 @@ def setup_parse
 
       @buf = ""
 
-      @params = Rack::Utils::KeySpaceConstrainedParams.new
+      @params = if rack_2x?
```

Comment:
We could make this method definition itself conditional based on the logic below. It won't change at runtime, and shaving off the check couldn't hurt